xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecAmlogic.cpp

```diff
@@ -216,6 +216,10 @@ bool CDVDVideoCodecAmlogic::Open(CDVDStreamInfo &hints, CDVDCodecOptions &option
     }
   }
 
+  m_processInfo.SetVideoDecoderName(m_pFormatName, true);
+  m_processInfo.SetVideoDimensions(m_hints.width, m_hints.height);
```

Comment:
sory, I dont follow

push it as is if you want, close if inappropriate

side note: https://github.com/stefansaraev/xbmc/commit/3c0cfa05a91a06db4ccdb811ce0d5fb37ac730b2 fixes reported resolution when sw decoding for me.